scripts/env-CMSRunAnalysis.sh

```diff
@@ -0,0 +1,136 @@
+    # - crab preparelocal, crab submit --dryrun: when running a job locally
+
+    # On some sites we know there was some problems with environment cleaning
+    # with using 'env -i'. To overcome this issue, whenever we start a job, we have
+    # to save full current environment into file, and whenever it is needed we can load
+    # it. Be aware, that there are some read-only variables, like: BASHOPTS, BASH_VERSINFO,
+    # EUID, PPID, SHELLOPTS, UID, etc.
+
+    # Moreover, src/python/WMCore/Storage/Backends/GFAL2Impl.py 
+    # makes use of startup_environment.sh
+
+    export DMDEBUGVAR=dmdebugvalue-env-cmsrunanalysis
+    export JOBSTARTDIR=$PWD
+    export HOME=${HOME:-$PWD}
+
+    declare -p > startup_environment.sh
+
+    # these lines are for debugging purposes only: start 
+    echo "DM DEBUG: cat startup_environment.sh"
+    # basename -- "$0"
+    # dirname -- "$0"
+    # echo $PWD
+    # ls -lrth
+    cat startup_environment.sh
+    # these lines are for debugging purposes only: end 
+
+}
+
+env_set_local () {
+    # when running a job locally, we need to set manually some variables that 
+    # are set for us when running on the global pool.
+
+    export SCRAM_ARCH=$(scramv1 arch)
+    export REQUIRED_OS=rhel7
+    export CRAB_RUNTIME_TARBALL=local
+    export CRAB_TASKMANAGER_TARBALL=local
+    export CRAB3_RUNTIME_DEBUG=True
+
+}
+
+env_cms_load() {
+    ### source the CMSSW stuff using either OSG or LCG style entry env. or CVMFS
+    echo "======== CMS environment load starting at $(TZ=GMT date) ========"
+    CMSSET_DEFAULT_PATH=""
+    if [ -f "$VO_CMS_SW_DIR"/cmsset_default.sh ]
+    then  #   LCG style --
+        echo "WN with a LCG style environment, thus using VO_CMS_SW_DIR=$VO_CMS_SW_DIR"
+        CMSSET_DEFAULT_PATH=$VO_CMS_SW_DIR/cmsset_default.sh
+    elif [ -f "$OSG_APP"/cmssoft/cms/cmsset_default.sh ]
+    then  #   OSG style --
+        echo "WN with an OSG style environment, thus using OSG_APP=$OSG_APP"
+        CMSSET_DEFAULT_PATH=$OSG_APP/cmssoft/cms/cmsset_default.sh CMSSW_3_3_2
+    elif [ -f "$CVMFS"/cms.cern.ch/cmsset_default.sh ]
+    then
+        echo "WN with CVMFS environment, thus using CVMFS=$CVMFS"
+        CMSSET_DEFAULT_PATH=$CVMFS/cms.cern.ch/cmsset_default.sh
+    elif [ -f /cvmfs/cms.cern.ch/cmsset_default.sh ]
+    then  # ok, lets call it CVMFS then
+        CVMFS=/cvmfs/cms.cern.ch
+        echo "WN missing VO_CMS_SW_DIR/OSG_APP/CVMFS environment variable, forcing it to CVMFS=$CVMFS"
+        CMSSET_DEFAULT_PATH=$CVMFS/cmsset_default.sh
+    else
+        echo "Error during job bootstrap: VO_CMS_SW_DIR, OSG_APP, CVMFS or /cvmfs were not found." >&2
+        echo "  Because of this, we can't load CMSSW. Not good." >&2
+        exit 11003
+    fi
+    . $CMSSET_DEFAULT_PATH
+    echo "export CMSSET_DEFAULT_PATH=$CMSSET_DEFAULT_PATH" >> startup_environment.sh
+    echo -e "========  CMS environment load finished at $(TZ=GMT date) ========\n"
+}
+
+load_comp_python() {
```

Comment:
why not `env_python_load` for consistency with `env_cms_load` above ?
BTW I'd rather put verb first in those names, e.g. : `save_env`, `setup_cms_env`, `setup_python`, `set_env_local`

Reply:
I always struggle with assigning names, thanks for pointing this out. I propose:

- `save_env`
- `setup_local_env` or `setup_env_local`, i can not decide what i prefer
- `setup_cmsset` (i realized I do not like `setup_cms_env`, too close to `cmsenv` which does something different from `cmsset_default.sh`)
- `setup_python_comp`